### DISCUSSION TITLE
Does on-demand ISR immediately trigger a rebuild or on the next load?

### DISCUSSION THREAD

**Discussion by rawcomposition** [vercel/next.js#44123] [Help] [answered]
When manually revalidating a statically rendered route, such as `res.revalidate('/products')`, does Nextjs immediately rebuild that page or simply mark it as stale and build it when a visitor next accesses the page?

**Comment by icyJoseph** [comment, marked as answer]
It builds it when you call `res.revalidate("/something")`, in fact, you can even `await` that call.
